gdc_client/query/versions.py

```diff
@@ -30,14 +30,23 @@ def get_latest_versions(url, uuids, verify=True):
         resp = requests.post(versions_url, json={"ids": chunk}, verify=verify)
 
         # Parse the results of the chunked query.
-        for result in resp.json():
-            file_id = result.get("id")
-            uuid = result.get("latest_id")
-            if uuid:
-                latest_versions[file_id] = uuid
-            else:
-                # Might happen for legacy files
-                latest_versions[file_id] = file_id
+        try:
```

Comment:
Is there a reason we opted not to check the status of the response to insure that we have a 2XX response before trying to deserialize the body? From the test that's what it looks like we are trying to solve for. If that is our main concern we could handle this with an if/else.

Reply:
i didn't find a good way to check if the response content is valid json: from what I've seen online, its suggested to just do a try/catch to see if its valid json. I was thinking about using the Response class's ok flag, which will return true if the response code is less than 400, but iti seemed safer to always just surround the json deserialization with try/catch. I'm ok though with just checking if the response code is 2xx or 3xx with the ok flag